server/controllers/search/search.js

```diff
@@ -32,21 +34,58 @@ const sanitization = {
   }
 }
 
-const search = async ({ types, search, lang, limit, filter, exact, minScore, claim, reqUserId }) => {
+const controller = async (params, req, res) => {
+  const { types, search, claim } = params
   if (!(_.isNonEmptyString(search) || _.isNonEmptyString(claim))) {
     throw error_.newMissing('query', 'search or claim')
   }
 
-  let results = await typeSearch({ lang, types, search, limit, filter, exact, minScore, claim })
+  const useSocialSearch = types.includes('users') || types.includes('groups')
 
-  results = results
+  if (useSocialSearch) {
+    return socialSearch(params, res)
+  } else {
+    return entitiesSearch(params)
+  }
+}
+
+const socialSearch = async (params, res) => {
+  const { search, lang, limit, offset, reqUserId } = params
+
+  if (!(_.isNonEmptyString(search))) {
+    throw error_.newMissing('query', 'search')
+  }
+
+  if (offset !== 0) {
+    addWarning(res, 'the offset parameter is ignored on user and group search')
+  }
+
+  const { hits } = await typeSearch(params)
+
+  const results = hits
     .filter(isSearchable(reqUserId))
     .map(normalizeResult(lang))
     .slice(0, limit)
 
   return { results }
 }
 
+const entitiesSearch = async params => {
+  const { search, lang, limit, offset, claim } = params
+
+  if (!(_.isNonEmptyString(search) || _.isNonEmptyString(claim))) {
+    throw error_.newMissing('query', 'search or claim')
+  }
+
+  const { hits, total } = await typeSearch(params)
+  const continu = limit + offset
```

Comment:
```suggestion
  const continue = limit + offset
```

Reply:
can't do: `continue` is a reserved word in JS ^^'